### ISSUE TITLE
Webhook URLs use masterURL instead of masterPublicURL

### ISSUE THREAD

**Issue by chuenlye** [openshift/origin#14406]
By command of `oc describe bc $name`, It seems that webhook URLs use masterURL instead of masterPublicURL.

Definition of masterURL and masterPublicURL are from the Origin's document:
https://docs.openshift.com/container-platform/3.5/install_config/master_node_configuration.html

In my environment, I have to  change the URLs from masterURL to masterPublicURL by hand.
MasterPublicURL seems always better than masterURL here, considering that the webhook is executed from outside of OpenShift.

##### Version
oc v1.5.0+031cbe4
kubernetes v1.5.2+43a9be4
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://${master_api_server}:8443
openshift v1.5.0+031cbe4
kubernetes v1.5.2+43a9be4

##### Steps To Reproduce
get webhook URLs by `oc describe bc <name>`

##### Current Result
masterURL is used in webhook URLs.

##### Expected Result
masterPublicURL will be used in webhook URLs.

**Comment by chuenlye** [comment]
I found that that was not a bug!

```
$ oc config get-clusters
NAME
${public_cluster_name_using_masterPublicURL}:8443
${local_cluster_name_using_masterURL}:8443
```

And,
```
$ oc config current-context
${project_name}/${local_cluster_name_using_masterURL}:8443/${user}
```

If the current-context use masterURL, webhook URL from `oc describe bc $name` will use masterURL.
If the current-context use masterPublicURL, webhook from `oc describe bc $name` will use masterPublicURL.

I used .kube/config copied from Origin master server after the fresh installation, 
while that .kube/config file specify current-context to masterURL's cluster.
So I met this problem. 

Sorry!
